diff --git a/js/mod.ts b/js/mod.ts
index 393c5d8..3db0663 100644
--- a/js/mod.ts
+++ b/js/mod.ts
@@ -17,7 +17,7 @@ export class ImportMap {
   }
 
   /** Given a specifier and a referrer, return the resolved string URL. */
-  resolve(specifier: string, referrer: string | URL) {
+  resolve(specifier: string, referrer: string | URL): string {
     if (referrer instanceof URL) {
       referrer = referrer.toString();
     }
